realm/realm-library/src/objectServer/java/io/realm/mongodb/AppConfiguration.java

```diff
@@ -338,6 +347,12 @@ public void onError(SyncSession session, AppException error) {
                 }
             }
         };
+        private SyncSession.ClientResetHandler defaultClientResetHandler = new SyncSession.ClientResetHandler() {
+            @Override
+            public void onClientReset(SyncSession session, ClientResetRequiredError error) {
+                RealmLog.error("Client Reset required for: " + session.getConfiguration().getServerUrl());
```

Comment:
It is the same behavior we have currently for the default behavior, but yes I was also considering if we should throw an exception instead. 

Throwing feels a bit heavy though because you can theoretically still use the app after this happens, Sync just stops working. If we throw, it might mean forcibly breaking _a lot_ of apps if the developer somehow failed to implement support. Without throwing there is a least a bigger chance that the app is still somewhat useful.